tavern/testutils/helpers.py

```diff
@@ -1,16 +1,59 @@
 import json
+import operator
 import importlib
 import logging
 import re
 import jwt
+import jmespath as jmes
+
 from box import Box
 
 from tavern.schemas.files import wrapfile, verify_generic
 
-
 logger = logging.getLogger(__name__)
 
 
+def test_type(val, mytype):
+    """ Check value fits one of the types, if so return true, else false """
+    typelist = TYPES.get(mytype.lower())
+    if typelist is None:
+        raise TypeError(
+            "Type {0} is not a valid type to test against!".format(mytype.lower()))
+    try:
+        for testtype in typelist:
+            if isinstance(val, testtype):
+                return True
+        return False
+    except TypeError:
+        return isinstance(val, typelist)
+
+COMPARATORS = {
+    'count_eq': lambda x, y: safe_length(x) == y,
+    'lt': operator.lt,
+    'less_than': operator.lt,
+    'eq': operator.eq,
+    'equals': operator.eq,
+    'str_eq': lambda x, y: operator.eq(str(x), str(y)),
+    'ne': operator.ne,
+    'not_equals': operator.ne,
+    'gt': operator.gt,
+    'greater_than': operator.gt,
+    'contains': lambda x, y: x and operator.contains(x, y),  # is y in x
+    'contained_by': lambda x, y: y and operator.contains(y, x),  # is x in y
+    'regex': lambda x, y: regex_compare(str(x), str(y)),
+    'type': test_type
+}
+TYPES = {
+    'none': type(None),
+    'number': (int, float),
+    'int': (int),
+    'float': float,
+    'bool': bool,
+    'str': str,
+    'list': list,
+    'dict': dict
+}
+
```

Comment:
Just a minor thing - could you move the extra functions/definitions (eg, `test_type`, `COMPARISONS`, `regex_compare`, etc) to a file `tavern/testutils/jmes.py` to keep it encapsulated?